.github/workflows/bats.yaml

```diff
@@ -50,42 +98,159 @@ jobs:
+      if: runner.os == 'Linux'
+      run: sudo chmod a+rwx /dev/kvm
+
+    - name: "Linux: Initialize pass"
+      if: runner.os == 'Linux'
       run: |
-        DEFAULT_DIR=$PWD
-        cd "$TMPDIR/bats"
-        ./bats-core/bin/bats \
-            --gather-test-outputs-in "$DEFAULT_DIR/logs" \
-            --report-formatter tap \
-            ${{ inputs.tests }}
+        # Configure the agent to allow default passwords
+        HOMEDIR="$(gpgconf --list-dirs homedir)" # spellcheck-ignore-line
+        mkdir -p "${HOMEDIR}"
+        chmod 0700 "${HOMEDIR}"
+        echo "allow-preset-passphrase" >> "${HOMEDIR}/gpg-agent.conf"
+
+        # Create a GPG key
+        gpg --quick-generate-key --yes --batch --passphrase '' \
+          user@rancher-desktop.test default \
+          default never
+
+        # Get info about the newly created key
+        DATA="$(gpg --batch --with-colons --with-keygrip --list-secret-keys)"
+        FINGERPRINT="$(awk -F: '/^fpr:/ { print $10 ; exit }' <<< "${DATA}")" # spellcheck-ignore-line
+        GRIP="$(awk -F: '/^grp:/ { print $10 ; exit }' <<< "${DATA}")"
+
+        # Save the password
+        gpg-connect-agent --verbose "PRESET_PASSPHRASE ${GRIP} -1 00" /bye
+
+        # Initialize pass
+        pass init "${FINGERPRINT}"
+
+    - name: Normalize test name
+      id: normalize
+      run: |
+        t="${{ matrix.name }}"
+        if [[ ! -r "tests/$t" ]] && [[ -r "tests/${t}.bats" ]]; then
+          t="${t}.bats"
+        fi
+        echo "test=$t" >> "$GITHUB_OUTPUT"
+      working-directory: bats
+
+    - name: "macOS: Set startup command"
+      if: runner.os == 'macOS'
+      run: echo "BATS_COMMAND=$BATS_COMMAND" >> "$GITHUB_ENV"
+      env:
+        BATS_COMMAND: exec
+    - name: "Linux: Set startup command"
+      if: runner.os == 'Linux'
+      run: echo "BATS_COMMAND=$BATS_COMMAND" >> "$GITHUB_ENV"
+      env:
+        BATS_COMMAND: >-
+          exec xvfb-run --auto-servernum
+          --server-args='-screen 0 1280x960x24'
+
+    - name: Run BATS
+      # We use ${{ env.BATS_COMMAND }} instead of ${BATS_COMMAND} to let the
+      # shell parse the command, instead of doing it via expansion which is then
+      # parsed differently (--server-args isn't kept as one word).
+      run: >-
+        ${{ env.BATS_COMMAND }}
+        ./bats-core/bin/bats
+        --gather-test-outputs-in "${LOGS_DIR}"
+        --formatter tap
+        --report-formatter tap
+        'tests/${{ steps.normalize.outputs.test }}'
       env:
         RD_CONTAINER_ENGINE:     ${{ matrix.engine }}
         RD_CAPTURE_LOGS:         "true"
         RD_TAKE_SCREENSHOTS:     "true"
-        RD_USE_IMAGE_ALLOW_LIST: "true"
+        RD_TRACE:                "true"
+        RD_USE_GHCR_IMAGES:      "true"
+        RD_USE_RAMDISK:          "true"
+        LOGS_DIR:                ${{ github.workspace }}/logs
+        GITHUB_TOKEN:            ${{ github.token }}
+        BATS_COMMAND:            ${{ env.BATS_COMMAND }}
```

Comment:
There is no need to define `BATS_COMMAND` if you are not using it in the `run` command.

Reply:
This actually exists because the VSCode GitHub Workflows checker thing falls over (on line 167) if it's not explicitly defined.  Note that I'm just defining it to itself here, since `env.BATS_COMMAND` must already exist…